server/model/index.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+// TODO: needs update when we integrate csv feature
+const data_report_params = schema.object({
+  saved_search_id: schema.string(),
+  time_range: schema.string(),
+  reportFormat: schema.oneOf([schema.literal('csv'), schema.literal('xlsx')]),
+});
+
+const visual_report_params = schema.object({
+  url: schema.uri(),
+  windowWidth: schema.number({ defaultValue: 1200 }),
+  windowLength: schema.number({ defaultValue: 800 }),
+  reportFormat: schema.oneOf([schema.literal('pdf'), schema.literal('png')]),
+});
+
+export const scheduleSchema = schema.object({
+  schedule_type: schema.oneOf([
+    schema.literal('Now'),
+    schema.literal('Future Date'),
+    schema.literal('Recurring'),
+    schema.literal('Cron Based'),
+  ]),
+  schedule: schema.any(),
+});
+
+export const interval = schema.object({
+  interval: schema.object({
+    period: schema.number(),
+    // Refer to job scheduler SPI https://github.com/opendistro-for-elasticsearch/job-scheduler/blob/b333469c183a15ddbf496a690300cc9e34d937fb/spi/src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/spi/schedule/IntervalSchedule.java
+    unit: schema.oneOf([
+      schema.literal('MINUTES'),
+      schema.literal('HOURS'),
+      schema.literal('DAYS'),
+    ]),
+    // timeStamp
+    start_time: schema.number(),
+  }),
+});
+
+export const cron = schema.object({
+  corn: schema.object({
+    expression: schema.string(),
+    time_zone: schema.string(),
+  }),
+});
+
+export const email = schema.object({
+  subject: schema.string(),
+  body: schema.string(),
+  has_attachment: schema.boolean({ defaultValue: true }),
+  recipients: schema.arrayOf(schema.string(), { minSize: 1 }),
+});
+
+export const reportSchema = schema.object({
+  report_name: schema.string(),
+  report_source: schema.oneOf([
+    schema.literal('Dashboard'),
+    schema.literal('Visualization'),
+    schema.literal('Saved Search'),
+  ]),
+  report_type: schema.oneOf([
+    schema.literal('Download'),
+    schema.literal('Alert'),
+    schema.literal('Schedule'),
+  ]),
+  description: schema.string(),
+  report_params: schema.conditional(
+    schema.siblingRef('report_source'),
+    'Saved Search',
+    data_report_params,
+    visual_report_params
+  ),
+
+  delivery: schema.maybe(
+    schema.object({
+      channel: schema.oneOf([
```

Comment:
Since we only have Kibana and Email as options right now, I think it should be ok to leave out Slack and Chime

Reply:
That's true, but it doesn't affect the schema definition here